### PR TITLE
Adding lead metadata to lead basic info card

### LINKED ISSUE #694: Show who created/modified leads on Lead homepage

### PR DESCRIPTION
**Pull request by TheMcMurder**
Fixes #694 

I chose to do it this way after examining other areas of the app. A contact has a similar header, it's slightly different but I tried to capture the look and feel with this change.

It didn't feel right in other areas and this area _felt_ the most visually consistent to me. If you want changes to the approach or location please let me know.

Screen Shot 2020-11-20 at 4 12 39 PM

### CODE REVIEW

File: frontend/view-edit-lead/lead-home/lead-metadata.component.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import dayjs from "dayjs";
+
+export default function LeadMetadata(props) {
```
Comment (TheMcMurder) [inline frontend/view-edit-lead/lead-home/lead-metadata.component.tsx:4]: Dumb typescript question.

Would it be better to type here and export or will it detect the typings from above?

What's better typings at the bottom of the component tree or typings at a higher level? I assume it's better at the bottom of the tree but I'm not really sure.

File: frontend/view-edit-lead/lead-home/lead-section.component.tsx [hunk 2]
```diff
@@ -43,6 +56,12 @@ const css = `
     font-size: 2.1rem;
   }
 
+  & .lead-metadata {
+    font-style: italic;
+    font-size: 1.2rem;
+    text-align: center;
+  }
+
```
Comment (TheMcMurder) [inline frontend/view-edit-lead/lead-home/lead-section.component.tsx:64]: Same styles as the `audit trail` https://github.com/JustUtahCoders/comunidades-unidas-internal/blob/master/frontend/view-edit-client/client-home/client-section.component.tsx#L86

File: frontend/view-edit-lead/lead-home/lead-section.component.tsx [hunk 3]
```diff
@@ -26,15 +26,28 @@ export default function LeadSection(props: LeadSectionProps) {
...
-      {expanded && <div className="lead-section-content">{props.children}</div>}
+      {expanded && <div className="lead-section-content">{children}</div>}
     </div>
   );
 }
 
 type LeadSectionProps = {
   title: string;
+  metadata?: {
+    createdBy: {
```
Comment (joeldenning) [inline frontend/view-edit-lead/lead-home/lead-section.component.tsx:40]: There is an existing type for this - `CUObjectAudit`. It's in `edit-invoice.component.tsx`

```suggestion
    createdBy: CUObjectAudit
```

**Review by joeldenning — COMMENTED** [review]
Looks good - one small comment

File: frontend/view-edit-lead/lead-home/lead-metadata.component.tsx (hunk 1 above)
Comment (joeldenning) [reply frontend/view-edit-lead/lead-home/lead-metadata.component.tsx:4]: best to type the component props:

```suggestion
export default function LeadMetadata(props: LeadMetadataProps) {
```

File: frontend/view-edit-lead/lead-home/lead-section.component.tsx (hunk 3 above)
Comment (TheMcMurder) [reply frontend/view-edit-lead/lead-home/lead-section.component.tsx:40]: Yes, I considered using that but it's not exactly the same so I wasn't sure if I should reuse it or not.

**Review by joeldenning — COMMENTED** [review]
Thanks!

**Pull request merged**